diff --git a/common/Kconfig b/common/Kconfig
index 1879aef..23b0a2e 100644
--- a/common/Kconfig
+++ b/common/Kconfig
@@ -180,6 +180,13 @@ config ENV_IS_IN_NAND
 	  Define this if you have a NAND device which you want to use for the
 	  environment.
 
+config ENV_IS_IN_SPI_FLASH
+	bool "Environment in a SPI flash"
+	depends on CMD_SF
+	help
+	  Define this if you have a SPI flash which you want to use for the
+	  environment.
+
 config ENV_IS_IN_UBI
 	bool "Environment in a UBI volume"
 	depends on CMD_UBI
@@ -211,6 +218,13 @@ config ENV_SIZE
 	help
 	  Size of the environment storage area
 
+config ENV_SECT_SIZE
+	hex "Flash sector size"
+	depends on ENV_IS_IN_SPI_FLASH
+	default 0x1000 if ARCH_SUNXI
+	help
+	  Size of the flash sector area
+
 config ENV_UBI_PART
 	string "UBI partition name"
 	depends on ENV_IS_IN_UBI
