From 8769e9010710a1e33dfb8f19ef7ae4667d8daff8 Mon Sep 17 00:00:00 2001
From: Jiri Novotny <jiri.novotny@logicelements.cz>
Date: Wed, 1 Feb 2017 14:44:01 +0100
Subject: [PATCH 8/8] fix makefile whitespace

Signed-off-by: Jiri Novotny <jiri.novotny@logicelements.cz>
---
 arch/arm/dts/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index d9de44c..06e8ced 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -272,7 +272,7 @@ dtb-$(CONFIG_MACH_SUN8I_A83T) += \
 	sun8i-a83t-sinovoip-bpi-m3.dtb
 dtb-$(CONFIG_MACH_SUN8I_H3) += \
 	sun8i-h2-plus-orangepi-zero.dtb \
-	sun8i-h3-bananapi-m2-plus.dtb \	
+	sun8i-h3-bananapi-m2-plus.dtb \
 	sun8i-h3-beelink-x2.dtb \
 	sun8i-h3-nanopi-m1.dtb \
 	sun8i-h3-nanopi-neo.dtb \
-- 
2.7.4

