From 14bda3e09947e539175ba7d9e126560a8c249c6b Mon Sep 17 00:00:00 2001
From: Jiri Novotny <jiri.novotny@logicelements.cz>
Date: Wed, 3 May 2017 21:10:56 +0200
Subject: [PATCH 2/2] add hope radio to spidev comapible list

Signed-off-by: Jiri Novotny <jiri.novotny@logicelements.cz>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 9e2e099..8537021 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -697,6 +697,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
 	{ .compatible = "ge,achc" },
+	{ .compatible = "hoperf,rfm7x" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

