diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fadf72..acace68 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ target_link_libraries (uWS LINK_PUBLIC ${ZLIB_LIBRARY})
 
 if (UNIX)
 target_link_libraries (uWS LINK_PUBLIC pthread)
-install (TARGETS uWS DESTINATION /usr/lib64)
+install (TARGETS uWS DESTINATION /usr/lib)
 install (FILES src/Extensions.h src/WebSocketProtocol.h src/Networking.h src/WebSocket.h src/Hub.h src/Group.h src/Node.h src/Socket.h src/HTTPSocket.h src/uWS.h DESTINATION /usr/include/uWS)
 endif (UNIX)
 
