--- a/mDNSPosix/Makefile
+++ b/mDNSPosix/Makefile
@@ -62,11 +62,6 @@
   os=linux
 endif
 
-CC = cc
-BISON = bison
-FLEX = flex
-ST = strip
-LD = ld
 SOOPTS = -shared
 CP = cp
 RM = rm
@@ -103,10 +98,22 @@
 CFLAGS_DEBUGGING = -g -DMDNS_DEBUGMSGS=0
 OBJDIR ?= objects/prod
 BUILDDIR ?= build/prod
-STRIP = $(ST) -S
 endif
 endif
 
+ifeq ($(os),buildroot)
+CFLAGS_OS =-DNOT_HAVE_SA_LEN -DUSES_NETLINK -DHAVE_LINUX -DTARGET_OS_LINUX -fno-strict-aliasing
+#OPTIONALTARG = nss_mdns
+#OPTINSTALL   = InstalledNSS
+else
+
+CC = @cc
+BISON = @bison
+FLEX = @flex
+ST = @strip
+LD = ld -shared
+STRIP = $(ST) -S
+
 # Configure per-OS peculiarities
 ifeq ($(os),solaris)
 CFLAGS_DEBUGGING = -O0 -DMDNS_DEBUGMSGS=0
@@ -186,6 +193,7 @@
 endif
 endif
 endif
+endif
 endif
 endif
 endif
